hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+            self.log.debug('running mempool sync, try again later')
+            return
+
+        bestblock = self.tx_storage.get_best_block()
+        meta = bestblock.get_metadata()
+
+        self.log.debug('run sync', height=meta.height)
+
+        assert self.protocol.connections is not None
+        assert self.tx_storage.indexes is not None
+        assert self.tx_storage.indexes.deps is not None
+
+        if self.tx_storage.indexes.deps.has_needed_tx():
+            self.log.debug('needed tx exist, sync transactions')
+            self.update_synced(False)
+            # TODO: find out whether we can sync transactions from this peer to speed things up
+            self.run_sync_transactions()
+        else:
+            # I am already in sync with all checkpoints, sync next blocks
+            yield self.run_sync_blocks()
+
+    def run_sync_transactions(self) -> None:
+        self.state = PeerState.SYNCING_TRANSACTIONS
+
+        assert self.protocol.connections is not None
+        assert self.tx_storage.indexes is not None
+        assert self.tx_storage.indexes.deps is not None
+
+        # start_hash = self.tx_storage.indexes.deps.get_next_needed_tx()
+        needed_txs, _ = collect_n(self.tx_storage.indexes.deps.iter_next_needed_txs(),
+                                  MAX_GET_TRANSACTIONS_BFS_LEN)
+
+        # Start with the last received block and find the best block full validated in its chain
+        block = self._last_received_block
+        if block is None:
+            block = cast(Block, self.tx_storage.get_genesis(settings.GENESIS_BLOCK_HASH))
+        else:
+            with self.tx_storage.allow_partially_validated_context():
+                while not block.get_metadata().validation.is_valid():
+                    block = block.get_block_parent()
+        assert block is not None
+        assert block.hash is not None
+        block_height = block.get_height()
+
+        self.log.info('run sync transactions', start=[i.hex() for i in needed_txs], end_block_hash=block.hash.hex(),
+                      end_block_height=block_height)
+        self.send_get_transactions_bfs(needed_txs, block.hash)
+
+    @inlineCallbacks
+    def run_sync_blocks(self) -> Generator[Any, Any, None]:
+        assert self.tx_storage.indexes is not None
+        self.state = PeerState.SYNCING_BLOCKS
+
+        # Find my height
+        bestblock = self.tx_storage.get_best_block()
+        assert bestblock.hash is not None
+        meta = bestblock.get_metadata()
+        my_height = meta.height
+
+        self.log.debug('run sync blocks', my_height=my_height)
+
+        # Find best block
+        data = yield self.get_peer_best_block()
+        peer_best_block = data['block']
+        peer_best_height = data['height']
+        self.peer_height = peer_best_height
+
+        # find best common block
+        yield self.find_best_common_block(peer_best_height, peer_best_block)
```

Comment:
I was just curious. Since this method doesn't return a `Deferred`, is the `yield` here actually doing anything?